Package vulkan-wsi-layer for Debian

haoyangw opened the pull request:
Changes:
* Generate basic Debian packaging configs to turn `vulkan-wsi-layer` into a Debian source package
* This generates `libvulkan-wsi-layer` after compilation using `dpkg-buildpackage`/`debuild`

haoyangw committed:
Debianise vulkan-wsi-layer
-Generate basic Debian packaging configs to turn vulkan-wsi-layer into a Debian source package
-This generates libvulkan-wsi-layer after compilation using dpkg-buildpackage/debuild

haoyangw committed:
Improve debian/copyright
-Include the full MIT license from ARM in the debian/copyright file to fulfil
 the specified terms(and for clarity since Debian states that there are
 multiple MIT licenses)

haoyangw commented on debian/copyright:
```diff
@@ -0,0 +1,30 @@
+Source: https://github.com/ginkage/vulkan-wsi-layer
+
+Files: *
+Copyright: 2019 Arm Limited
```
@ginkage Other than ARM I'm not sure who are the specific contributors from your end for Rockchip, so I leave this part to you to add contributors if there're anyone important to credit alright?

ginkage replied:
Nah, it's just me. I think it's fine to leave just ARM.

haoyangw commented on debian/rules:
```diff
@@ -0,0 +1,26 @@
+		       -DCMAKE_INSTALL_PREFIX=/usr \
+                       -DSELECT_EXTERNAL_ALLOCATOR=dma_buf_heaps \
+                       -DWSIALLOC_MEMORY_HEAP_NAME=system-uncached \
+                       -DKERNEL_HEADER_DIR=/usr/src/linux-headers-*-rk35xx/arch/$(DEB_BUILD_ARCH)/include \
```
Potential issue I can think of is if someone can both the `legacy` and `vendor` Armbian header packages installed on the same system when they compile the package, but this seems like an edge case that I'm unsure how to handle so it's neglected for now

ginkage replied:
Actually, you don't need headers at all, and this parameter can be removed. :)

haoyangw replied:
I see, done in the latest commit :) thanks

haoyangw commented on debian/changelog:
```diff
@@ -0,0 +1,6 @@
+vulkan-wsi-layer (1.3.276-1) unstable; urgency=medium
```
I've set the version for the package to the Vulkan version exposed by libmali, but perhaps there's a more accurate version to set for the package?

ginkage replied:
Doesn't really matter, I guess. Could even be 1.0.

ginkage commented on debian/control:
```diff
@@ -0,0 +1,36 @@
+ libwayland-bin [linux-any],
+ libwayland-client0 [linux-any],
+ libwayland-dev [linux-any], 
+ linux-headers-legacy-rk35xx | linux-headers-vendor-rk35xx [linux-any],
```
So yeah, headers aren't needed.

haoyangw replied:
Removed in the latest commit, thanks!

haoyangw committed:
Remove unneeded dependency
-Kernel headers are not required, so remove the dependency from
 the Debian package and also remove the kernel header cmake build flag

ginkage commented:
Nice, thanks a lot!
